src/js/tech/middleware.js

```diff
@@ -53,6 +77,14 @@ function middlewareIterator(method) {
   };
 }
 
+function executeRight(mws, method, value, terminated) {
+  for (let i = mws.length - 1; i >= 0; i--) {
+    const mw = mws[i];
+
+    mw[method](value, terminated);
```

Comment:
can we swap the argument order so it's similar to a node-style callback?

Reply:
Good idea, we can do that.